src/features/Form/Field/typing.ts

```diff
@@ -15,11 +15,11 @@ export type TField = {
   asterisk?: boolean;
   errorMessage?: string;
 } & TReactHookFormInputElement &
-  TFieldConfig;
+  TFieldConfig<string>;
 
-export type TFieldConfig = {
+export type TFieldConfig<T> = {
```

Comment:
```suggestion
export type TFieldConfig<T = string> = {
```
Чтобы была обратная совместимость и не пришлось везде переписывать.